include/alpaka/block/shared/dyn/BlockSharedMemDynMember.hpp

```diff
@@ -74,14 +74,15 @@ namespace alpaka
                      */
                     unsigned int staticMemCapacity() const
                     {
-                        return staticAllocBytes - m_dynPitch;
+                        return staticAllocBytes() - m_dynPitch;
                     }
 
                     //! Storage size in bytes
-                    static constexpr unsigned int staticAllocBytes = TStaticAllocKiB<<10;
+                    static constexpr unsigned int staticAllocBytes() {return TStaticAllocKiB<<10;}
 
                 private:
-                    mutable std::array<uint8_t, staticAllocBytes> m_mem;
+
+                    mutable std::array<uint8_t, staticAllocBytes()> m_mem;
```

Comment:
MSVC is complaining about this line. With the the function `staticAllocBytes` above only containing a single expression, this is valid C++11 and should not even require C++14.

Reply:
I agree it should be working even with C++11. I can take a look locally with my MSVS, but probably some time tomorrow.

Reply:
This seems like a VS 2017 bug. A workaround is to explicitly specify the class name for the static method call: instead of `staticAllocBytes()` put `BlockSharedMemDynMember<TStaticAllocKiB>::staticAllocBytes()`. At least on my machine this piece compiles then.

Reply:
Should we just do that then and add a comment, that the class name is given because of a bug in MSVC 2017?

Or should there be compile-time switch testing for MSVC 2017? In this case: how do I check for this MSVC version?

Reply:
I think here the lesser evil is to just put type name always and add a comment.
In case you want to go for an explicit test, can use `_MSC_VER` for detection. [Here is a table with values for different MSVS versions](https://dev.to/yumetodo/list-of-mscver-and-mscfullver-8nd)

Reply:
Addup: to test, one needs smth like `#if defined(_MSC_VER) && _MSC_VER >= NUMBER_FROM_THE_TABLE`

Reply:
I chose the lesser evil. Thanks.